vertx-web/src/main/java/io/vertx/ext/web/impl/HeaderParser.java

```diff
@@ -0,0 +1,156 @@
+   * @return The list of (unparsed) parsable header value
+   */
+  public static <T extends ParsedHeaderValue> List<T> convertToParsedHeaderValues(String unparsedHeaderValue,
+      Function<String, T> objectCreator){
+    
+    if(unparsedHeaderValue == null){
+      return Collections.emptyList();
+    } else if(unparsedHeaderValue.length() > MAX_HEADER_SIZE){
+      throw new HeaderTooLongException("Header longer than " + MAX_HEADER_SIZE + " characters");
+    }
+    String[] listedMIMEs = COMMA_SPLITTER.split(unparsedHeaderValue);
+    List<T> parsedMIMEs = new ArrayList<>(listedMIMEs.length);
+    for (String listedMIME : listedMIMEs) {
+      parsedMIMEs.add(objectCreator.apply(quotesRemover(listedMIME)));
+    }
+    return parsedMIMEs;
+  }
+  
+  /**
+   * In-place sorting of the headers list
+   * @param headers
+   * @return The same object as inserted
+   */
+  @Fluent
+  public static <T extends ParsedHeaderValue> List<T> sort(List<T> headers){
+    Collections.sort(headers, HEADER_SORTER);
+    return headers;
+  }
+  
+  static String quotesRemover(String val){
+    return val.replace("\\\"", "\"");
+  }
+  
+  static String matchedSelector(String... matches){
+    for (String match : matches) {
+      if(match != null){
+        return match;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Parses a header value
+   * 
+   * @param headerContent 
+   * @param valueCallback
+   * @param weightCallback
+   * @param parameterCallback
+   */
+  public static void parseHeaderValue(
+        String headerContent,
+        Consumer<String> valueCallback,
+        Consumer<Float> weightCallback,
+        BiConsumer<String, String> parameterCallback
+      ) {
+    int paramIndex = headerContent.indexOf(';');
+    
+    if(paramIndex < 0){
+      valueCallback.accept(headerContent);
+    } else {
+      valueCallback.accept(headerContent.substring(0, paramIndex));
+
+      Matcher paramFindings = PARAMETER_FINDER.matcher(headerContent);
+      
+      while(paramFindings.find()){
+        String key = paramFindings.group("key");
+        String value = matchedSelector(paramFindings.group("value1"), paramFindings.group("value2"));
+        // If "q" doesn't have a double as a value, it is ignored on purpose!
+        if("q".equalsIgnoreCase(key)){
+          try{
+            if(value != null){
+              weightCallback.accept(Float.parseFloat(value));
+            }
+          }catch(NumberFormatException e){
+            log.info("Found a \"q\" parameter with value \"{}\" which was unparsable", value);
+          }
+        } else {
+          parameterCallback.accept(key, value);
+        }
+      }
+    }
+  }
+  
+  public static void parseMIME(
+        String headerContent,
+        Consumer<String> componentCallback,
+        Consumer<String> subcomponentCallback
+      ){
+
+    int slashIndex = headerContent.indexOf('/');
+    int paramIndex = headerContent.indexOf(';', slashIndex);
```

Comment:
yes I saw that after, it would look more correct to me as we know that paramIndex > slashIndex since slashIndex == -1 or slashIndex is the index of `/` that is not `;`